R/Spectra-functions.R

```diff
@@ -38,26 +59,63 @@ addProcessing <- function(object, FUN, ...) {
 #'
 #' @param queue `list` of `ProcessStep` elements.
 #'
+#' @param spectraData `data.frame` with additional spectra variables as
+#'     requested with `addProcessing`.
+#'
 #' @return `list` of peak `matrix` elements.
 #'
 #' @author Johannes Rainer
 #'
 #' @importFrom ProtGenerics executeProcessingStep
 #'
 #' @noRd
-.apply_processing_queue <- function(x, msLevel, centroided, queue = NULL) {
+.apply_processing_queue <- function(x, spectraData = data.frame(),
+                                     queue = NULL) {
     if (length(queue)) {
+        args <- list()
+        ls <- length(spectraData)
+        if (ls)
+            colnames(spectraData) <- sub("^msLevel$", "spectrumMsLevel",
```

Comment:
Why do we need that renaming? Can't we not just use `msLevel` instead of `spectrumMsLevel`?

Reply:
I now guess the reason is that `msLevel` refers to the level on which the processing is applied on and `spectrumMsLevel` the actual level of the spectra (from `spectraVariables`).

Reply:
Exactly. We basically wanted to be consistent with `MSnbase` and allow the user to specify the MS level on which to apply the function with parameter `msLevel`. That's why we renamed the spectrum's MS level variable to `spectrumMsLevel`.